src/Compilers/VisualBasic/Portable/Operations/VisualBasicOperationFactory.vb

```diff
@@ -369,15 +369,15 @@ Namespace Microsoft.CodeAnalysis.Operations
 
         Private Function CreateBoundCallOperation(boundCall As BoundCall) As IInvocationOperation
             Dim targetMethod As IMethodSymbol = boundCall.Method
-            Dim receiver As IOperation = Create(boundCall.ReceiverOpt)
 
-            Dim instance As Lazy(Of IOperation) = New Lazy(Of IOperation)(Function() If(targetMethod.IsShared, Nothing, receiver))
+            Dim instance As Lazy(Of IOperation) = New Lazy(Of IOperation)(
+                    Function() CreateReceiverOperation(If(boundCall.ReceiverOpt, boundCall.MethodGroupOpt?.ReceiverOpt), targetMethod))
             Dim isVirtual As Boolean =
                 targetMethod IsNot Nothing AndAlso
                 instance IsNot Nothing AndAlso
                 (targetMethod.IsVirtual OrElse targetMethod.IsAbstract OrElse targetMethod.IsOverride) AndAlso
-                receiver.Kind <> BoundKind.MyBaseReference AndAlso
-                receiver.Kind <> BoundKind.MyClassReference
+                If(boundCall.ReceiverOpt?.Kind <> BoundKind.MyBaseReference, False) AndAlso
+                If(boundCall.ReceiverOpt?.Kind <> BoundKind.MyClassReference, False)
```

Comment:
Consider using `expr = True` rather than `If(expr, False)`.

Reply:
In VB this probably won't work because it uses three valued logic.

---
In reply to: [149578244](https://github.com/dotnet/roslyn/pull/22972#discussion_r149578244) [](ancestors = 149578244)